### PR TITLE
Bad signature fix and missing tests

### PR DESCRIPTION
**Pull request by kshyatt**
Pretty sure this wasn't supposed to be `collect`...

### CODE REVIEW

File: stdlib/LinearAlgebra/test/symmetric.jl [hunk 1]
```diff
@@ -592,4 +596,11 @@ end
     @test Symmetric(B) == Symmetric(Matrix(B))
 end
 
+@testset "symmetric()/hermitian() for Numbers" begin
+    @test symmetric(1, :U) == 1
+    @test symmetric_type(Int) == Int
+    @test hermitian(1, :U) == 1
+    @test hermitian_type(Int) == Int
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/test/symmetric.jl]: These four commands are not exported, and require specialization with `LinearAlgebra.symmetric` etc.

**Review by dkarrasch — APPROVED** [review]
That should fix CI.

File: stdlib/LinearAlgebra/test/symmetric.jl (hunk 1 above)
Comment (kshyatt) [reply stdlib/LinearAlgebra/test/symmetric.jl]: Derp, missed that somehow. Thanks!

**Comment by kshyatt** [comment]
Good to go?

**Pull request merged**